src/database/DataTypes/StocktakeBatch.js

```diff
@@ -115,11 +115,45 @@ export class StocktakeBatch extends Realm.Object {
   }
 
   /**
-   * Returns if this batch has a reason applied. Simple getter
-   * matching StocktakeItem interface for ease of use.
+   * Indicator that this batch has a positive difference in snapshot and
+   * counted quantity.
    */
-  get hasReason() {
-    return !!this.option;
+  get hasPositiveAdjustment() {
+    return this.difference > 0;
+  }
+
+  /**
+   * Indicator that this batch has a negative difference in snapshot and
+   * counted quantity.
+   */
+  get hasNegativeAdjustment() {
+    return this.difference < 0;
+  }
+
+  /**
+   * Returns an indicator that this batches reason/option state is valid.
+   * Valid being: negative differences require a negativeInventoryAdjustment
+   * option. Positive differences require a positiveInventoryAdjustment option
+   * while no difference requires there to be no option applied.
+   */
+  get validateReason() {
+    const { difference, option, hasPositiveAdjustment, hasNegativeAdjustment } = this;
+
+    // Short circuits for simple cases
+    if (!difference && !option) return true;
+    if (difference && !option) return false;
```

Comment:
I think the explicitness is good, if you actually have to follow this logic with the amount of states for such a simple thing a little bit of help for no real cost is what I am thinking

Reply:
Fair enough. I'd suggest that in this case you can still retain the explicitness and still make it a bit clearer (at least is clearer for me):

```
if (!option) {
  if (!difference) return true;
  if (difference) return false;
}
```

I think reducing duplication actually makes the code easier to follow, this way you're only maintaining three constraints in your head instead of four, and it also removes all the logical operations 🤷‍♂ .